benchmarks/benchmark_event_db.cpp

```diff
@@ -0,0 +1,94 @@
+#include <viewer/event_db.hpp>
+#include <viewer/base_ui.hpp>
+#include <hawktracer.h>
+#include <benchmark/benchmark.h>
+#include <hawktracer/parser/file_stream.hpp>
+#include <hawktracer/parser/protocol_reader.hpp>
+#include <hawktracer/parser/make_unique.hpp>
+
+using HawkTracer::parser::WellKnownKlasses;
+
+static void generate_test_tracepoints(HT_Timeline* timeline, int tracepoints)
+{
+    for (int i = 0; i < tracepoints; i++)
+    {
+        HT_DECL_EVENT(HT_Event, event);
+        ht_timeline_push_event(timeline, &event);
+    }
+}
+
+static void handle_event(const HawkTracer::parser::Event& event, HawkTracer::viewer::EventDB& event_db, HawkTracer::viewer::TimeRange& total_ts_range)
+{
+    const auto& klass = event.get_klass();
+
+    switch(static_cast<WellKnownKlasses>(klass->get_id()))
+    {
+        case WellKnownKlasses::EventKlassInfoEventKlass:
+        case WellKnownKlasses::EventKlassFieldInfoEventKlass:
+            break;
+        default:
+            event_db.insert(event);
+            auto ts = event.get_timestamp();
+            if (ts < total_ts_range.start)
+            {
+                total_ts_range.start = ts;
+            }
+            if (ts > total_ts_range.stop) 
+            {
+                total_ts_range.stop = ts;
+            }
+    }
+}
+
+static void BenchmarkEventDbGetData(benchmark::State& state)
+{
+    // Create timeline
+    HT_Timeline timeline;
+    ht_timeline_init(&timeline, 1024, HT_FALSE, HT_FALSE, nullptr);
+
+    HT_FileDumpListener file_dump_listener;
+    HT_ErrorCode error_code = ht_file_dump_listener_init(&file_dump_listener, "test_output.htdump", 4096u);
+    if (error_code != HT_ERR_OK)
+    {
+        char buffer[100];
+        sprintf(buffer, "Failed to initialize file dump listener! Error code: %d", error_code);
```

Comment:
I know that in your case it's not a problem, but in general it's always better to use `snprintf` instead. If someone changes the error message, and doesn't notice the buffer size, the app will crash.